fix: access-token does not get refreshed if not present

macrozone opened the pull request:
fixes https://github.com/kubernetes-client/javascript/issues/178

it does not, however, store the access-token back into .kube/config, i think that is a separate issue.

macrozone commented:
@brendandburns ok done. Please consider adding a prettier-config to the project.

brendandburns commented on src/cloud_auth.ts:
```diff
@@ -7,51 +7,69 @@ import { User } from './config_types';
+        return 'Bearer ' + config['access-token'];
+    }
+
+    private isExpired(config) {
```
Add type information here.

macrozone replied:
agree, but there is no type definition for `authProvider`, it's `any`-type, so i would have to invent a new type for the config.

brendandburns commented on src/cloud_auth.ts:
```diff
@@ -7,51 +7,69 @@ import { User } from './config_types';
+    private isExpired(config) {
+        const token = config['access-token'];
         const expiry = config.expiry;
+        if (!token) {
```
nit: inline `config['access-token']` or use `contains`

macrozone replied:
would break the symetry here between `access-token` and `expiry`.

i can inline it if you really want, but it does not make the code any better imho

brendandburns commented on src/cloud_auth.ts:
```diff
@@ -7,51 +7,69 @@ import { User } from './config_types';
-                } else {
-                    throw new Error('Token is expired!');
-                }
+    private updateAccessToken(config) {
```
Include type information for the parameter.

macrozone replied:
see above

brendandburns commented:
Some small comments, also the commit history seems messed up here. Needs a rebase?

macrozone commented:
@brendanburns i rebased, that's why the history is messed up. I usually don't rebase as it is very error prone and i am not used to it. I have no idea how i can fix that.

macrozone commented:
I added a type for config and fixed a bug. 
The tslint forces an interface for that type (i come from flow-type, where you usually just use `type` for that). Does it look correct?

brendandburns commented:
This looks fine, but CI/CD is failing. run `npm test`

brendandburns commented:
@macrozone friendly ping on this if you want to get it to merge...

macrozone commented:
@brendanburns on it

brendandburns commented:
/lgtm
/approve
